tasks/db_deploy/db_deploy.py

```diff
@@ -1,387 +1,212 @@
-OS_ENVIRON_PLATFORM_KEY = 'PLATFORM'
-OS_ENVIRON_DDL_DIR_KEY = 'DDL_DIR'
-OS_ENVIRON_DROP_DATABASE_KEY = 'DROP_DATABASE'
-
-
-class DatabaseError(Exception):
-    """
-    Exception to be raised when there's a database error.
-    """
-
-
-def task() -> None:
-    """
-    Task called by the handler to perform the work.
-
-        Environment Vars:
-            Same as Handler.
-            TODO. Ideally move them into handler so this can be pure params
-
-    Raises:
-        DatabaseError: An error occurred.
-    """
-    prefix = os.environ['PREFIX']
-
-    secretsmanager = boto3.client('secretsmanager')
-    parameter = secretsmanager.get_secret_value(SecretId=f"{prefix}-drdb-user-pass")
-    db_user_pw = parameter['SecretString']
-
-    parameter = secretsmanager.get_secret_value(SecretId=f"{prefix}-drdb-admin-pass")
-    master_user_pw = parameter['SecretString']
+# Imports
+from shared_db import logger, get_configuration, get_root_connection
+from sqlalchemy import text
+from create_db import create_fresh_orca_install
+from migrate_db import perform_migration
+from typing import Any, Dict
 
-    parameter = secretsmanager.get_secret_value(SecretId=f"{prefix}-drdb-host")
-    db_host = parameter['SecretString']
 
-    drop_database = os.environ.get(OS_ENVIRON_DROP_DATABASE_KEY, 'False')
-    drop_database = (drop_database == 'True')
+# Globals
+# Latest version of the ORCA schema.
+LATEST_ORCA_SCHEMA_VERSION = 2
 
-    inner_task(db_host,
-               os.environ[OS_ENVIRON_DATABASE_NAME_KEY],
-               os.environ[OS_ENVIRON_DATABASE_PORT_KEY],
-               os.environ[OS_ENVIRON_DATABASE_USER_KEY],
-               db_user_pw, master_user_pw, drop_database)
 
-
-def inner_task(db_host: str, db_name: str, db_port: str, db_user: str, db_user_pass: str, db_admin_pass: str,
-               drop_database: bool) -> None:  # todo: Once large tests are adjusted/dropped, consider code reorg.
+def handler(
+    event: Dict[str, Any], context: object
+) -> None:  # pylint: disable-msg=unused-argument
     """
-    Task called by the handler to perform the work.
+    Lambda handler for db_deploy. The handler generates the database connection
+    configuration information, sets logging handler information and calls the
+    Lambda task function. See the `get_configuration()` function for
+    information on the needed environment variables and parameter store names
+    required by this Lambda.
 
     Args:
-        db_host: The database host.
-        db_name: The name of the database.
-        db_port: The database port.
-        db_user: The name of the application user.
-        db_user_pass: The password for the application user to connect to the database with.
-        db_admin_pass: An admin level password to connect to the database with.
-        drop_database: When true, will execute a DROP DATABASE command.
-
-    Returns:
-        string: description of status.
+        event (Dict): Event dictionary passed by AWS.
```

Comment:
I don't think you need to write out the type here since it is locked down in params.